### ISSUE TITLE
Logstash error after modifying default logstash.yml

### ISSUE THREAD

**Issue by dtrizna** [elastic/logstash#6545]
I'm observing strange Logstash behavior on my Centos 7 machine.

I've faced the problem, when in some moments after starting logstash via systemctl it displayed as active service without any errors in log file, but in same time it didn't start to listen on TCP/9600 and doesn't loaded config from conf.d.

For now I'm understand, that it was worked only with default configuration. Any time I changed something in /etc/logstash/logstash.yml it fails to start.

This problem is even if I try to launch binary file itself.
Now I'm reinstalled Logstash completely. There's two config files under conf.d. 
Here's representation:
[root@l-netflow1 logstash]# bin/logstash -t --path.settings /etc/logstash
Sending Logstash's logs to /var/log/logstash which is now configured via log4j2.properties
Configuration OK

[root@l-netflow1 logstash]# vim /etc/logstash/logstash.yml
 http.host: "127.0.0.1"  <== Uncomment this line f.e. and :wq

[root@l-netflow1 logstash]# bin/logstash -t --path.settings /etc/logstash
ERROR: Failed to load settings file from "path.settings". Aborting... path.setting=/etc/logstash, exception=Psych::SyntaxError, message=>(<unknown>): expected <block end>, but found BlockMappingStart while parsing a block mapping at line 138 column 2

If I rollback change by commenting line back, its again working fine.
As a result with any change in logstash.yml service via systemctl isn't starting (despite that systemd displaying it's ok). It simply says nothing in logs..

Info about system and Logstash itself:
[root@l-netflow1 logstash]# uname -a
Linux l-netflow1 3.10.0-514.2.2.el7.x86_64 #1 SMP Tue Dec 6 23:06:41 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
[root@l-netflow1 logstash]# curl -XGET "http://localhost:9600/?pretty=true"
{
  "host" : "l-netflow1.energo.lv",
  "version" : "5.1.2",
  "http_address" : "127.0.0.1:9600",
  "id" : "921e9e83-f2bd-4443-9e33-0d3022358662",
  "name" : "l-netflow1.energo.lv",
  "build_date" : "2017-01-11T20:36:32+00:00",
  "build_sha" : "1da2b64c9ca1df74cba80ae8fe4f476e69578733",
  "build_snapshot" : false

I think something similar was with guys in these posts:
https://discuss.elastic.co/t/logstash-5-alpha5-1-failed-to-load-settings-file-from-path-settings-aborting/58360
http://stackoverflow.com/questions/38799162/how-to-start-logstash-via-systemctl-on-centos7

**Comment by jsvd** [comment]
@dtrizna can you post more information on the error you received and the modified logstash.yml?

**Comment by dtrizna** [comment]
jvsd, accidentaly pressed enter before described whole info, now it's updated.

**Comment by jsvd** [comment]
@dtrizna this usually comes down to an invalid yaml file
please check the yaml is well formed, in your case there's probably a space before http.host

**Comment by dtrizna** [comment]
**A lot of uncensored words**
Didn't think that 'space' can do so much troubles..

@jsvd, this really was the case! Thank you for help!!

**Comment by jsvd** [comment]
This feedback definitely needs improvement, like you said there are others hitting the same issue (another example https://github.com/elastic/logstash/issues/6543)

**Comment by jsvd** [comment]
for tracking purposes, this is a dup of https://github.com/elastic/logstash/issues/6509

**Comment by dtrizna** [comment]
Its good that you're thinking in this direction.
My suggestions:
1) Of course problem could be solved completely if LS would ignore that space in the beginning of line (does ES/Kibana have the same behaviour?);

2) Possibly I would spent less time on troubleshooting, if exception, that triggers after launch of binary file, would be more precise, because the original one doesn't say anything useful for usual admin (syntax error, ok, but where?).

3) In addition to previous, it would be nice to have some 'Info' level log message. I even couldn't apply debug level logging in yml file due to problem itself.

**Comment by jsvd** [comment]
I created [this PR](https://github.com/elastic/logstash/pull/6546/files) to improve the messaging on a failed yaml parse. What do you think?

**Comment by dtrizna** [comment]
Looks nice! Definitely this is more precise information.
Will see in closest future does this help to mitigate such kind of problems.

Good job!

**Comment by hackery** [comment]
(2 years later ...) I suspect there are still a number of edge cases in yaml parsing, some of which may be deep inside Psych and not easily fixable except upstream.

We had a situation a couple of weeks ago after upgrading to 7.1.1 where the monitoring pipeline consistently refused to start; no useful log output until at debug level I noticed that the initial config summary didn't even include the `xpack.monitoring.*` entries.

After further trial and error we spotted that the entire `logstash.yml` had one initial space before all settings (I think someone had wanted to retain alignment when entries were #'d out) but the monitoring entries, added using automation, were not. Parsing seems to have been happily accepting the one-char indent as the root level, but then not handling the outdent, even with an error message.
